api/src/main/java/io/grpc/NameResolver.java

```diff
@@ -545,16 +561,169 @@ public Builder setOverrideAuthority(String authority) {
         return this;
       }
 
+      /**
+       * See {@link Args#getExtension(Key)}.
+       *
+       * <p>Optional, {@link Extensions#EMPTY} by default.
+       */
+      public Builder setExtensions(Extensions extensions) {
```

Comment:
Completely removed Extensions from the public API (it's now @Internal)

I tried maintaining a map in ManagedChannelImplBuilder but the code to copy entries over was problematic because 1) one more place needing unchecked casts 2) it's doing work that's best encapsulated and unit tested inside Args.Extensions.

I also tried out moving the NameResolver.Args.Builder to ManagedChannelImplBuilder. This worked but only because the current ManagedChannelImpl ctor unconditionally sets every property of Args.Builder. This is fragile - if this code ever changed to conditionally set a property it would silently leak that property across channels  when the builder is reused. This seemed too fragile to me.